src/builders/AvailabilityCalendarModelBuilder.ts

```diff
@@ -0,0 +1,55 @@
+import { AvailabilityStatus, CapabilityId, PricingPer } from "@octocloud/types";
+import { PricingDataProvider } from "../dataProviders/PricingDataProvider";
+import { DateHelper } from "../helpers/DateHelper";
+import { AvailabilityCalendarData } from "../data/AvailabilityCalendarData";
+import { AvailabilityCalendarModel } from "../models/Availability/AvailabilityCalendarModel";
+import { AvailabilityCalendarPricingModel } from "../models/Availability/AvailabilityCalendarPricingModel";
+import { AvailabilityCalendarPricingModelFactory } from "../factories/AvailabilityCalendarPricingModelFactory";
+
+interface AvailabilityCalendarModelBuilderData {
+  availabilityCalendarData: AvailabilityCalendarData;
+  pricingPer?: PricingPer;
+  capabilities?: CapabilityId[];
+}
+
+const defaultPricingPer: PricingPer = PricingPer.UNIT;
+const defaultCapabilities: CapabilityId[] = [CapabilityId.Pricing];
+
+export class AvailabilityCalendarModelBuilder {
+  public build(builderData: AvailabilityCalendarModelBuilderData): AvailabilityCalendarModel {
+    builderData.pricingPer ??= defaultPricingPer;
+    builderData.capabilities ??= defaultCapabilities;
+
+    const availabilityCalendarData = builderData.availabilityCalendarData;
+    const date = new Date();
+
+    return new AvailabilityCalendarModel({
+      localDate: availabilityCalendarData.localDate ?? DateHelper.formatForAvailability(date),
+      available: availabilityCalendarData.available ?? false,
+      status: availabilityCalendarData.status ?? AvailabilityStatus.CLOSED,
+      vacancies: availabilityCalendarData.vacancies ?? 0,
+      capacity: availabilityCalendarData.capacity ?? 20,
+      openingHours: availabilityCalendarData.openingHours ?? [
```

Comment:
it's better to default it to the empty array

there are 2 types of availability:
1. start time - don't have opening hours
2. opening hours - this one does